woocommerce/woocommerce-ios
# Shipping Label: Add customs row for international shipping label form

Closes #4688 

# Description
This PR updates the shipping label creation form to add a new row for customs form when international shipping is detected. There are a few things to be considered:
- Customs row should be inserted / removed when requirement for customs form changes. E.g: when destination address is updated so that shipping is within the US only, the row should be removed.
- When either address is updated so that customs form is required, there should be a check to make sure that phone number of origin address is present; if not - origin address row should be updated to pending and the subsequent rows should be updated to make sure that only 1 pending row is editable at a time.

# Changes
- Added new Row case `.customs` and configured table view cell for this type.
- Updated UIImage with new globe icon for use in the customs cell
- Updated ShippingLabelFormViewModel:
  - Made `generateInitialSections` method non-static, let it check for `customsFormRequired` to insert customs cell if needed
  - Added new private method `updateRowsForCustomsIfNeeded` to update origin address row to pending state if customs form is required and phone number for origin address is missing. This method is called whenever there's an update to either origin or destination address, since both these addresses can affect value of `customsFormRequired`.
  - Updated method `updateRowState` to insert / remove customs row when necessary. Also added a check to make sure that there's only 1 pending row editable at once.
- ShippingLabelFormViewController was also updated to make sure that it navigates to edit address form if phone number is missing (disregard of whether the address is validated or not).

# Demo
![customs-row](https://user-images.githubusercontent.com/5533851/128159647-1426618c-e271-4fa4-873e-f3e5bd81ab62.gif)

# Testing steps
1. Make sure that your test store is eligible for creating shipping labels (has WCShip plugin installed).
2. Navigate to Orders tab and select an order with status Processing.
3. Select Create Shipping Label.
4. If the order is international (origin country different from destination country, or one of the addresses is one of [US military states](https://git.io/JBuja)), notice that customs row is present. Otherwise, customs row should not be present.
5. Suppose the order is international, update the destination country to US and state being non-military state, proceed and notice that customs rows has been removed.
6. Update destination row to an international address, proceed and notice that customs row is added to the form.
7. If the origin address doesn't has a phone number, notice that the Ship From row is pending and editable, while other rows are not. Tapping on the Ship From row should navigate to edit address screen with a validation error about missing phone number.

Update release notes:
